enhancements/microshift/microshift-updateability-ostree.md

```diff
@@ -0,0 +1,1156 @@
+     > `current-commit not in health.json` &&
+     > `prev-boot-commit != ostree rollback commit`
+1. `microshift run`
+1. System is unhealthy, greenboot reboots the system (3rd commit)
+1. `microshift pre-run`
+   - Delete data and start clean
+     > `prev-boot-commit.system == unhealthy` &&
+     > `current-commit == prev-boot-commit` &&
+     > `backups/current-commit.id/ does not exist` &&
+     > `earlier-commit in health.json` &&
+     > `earlier-commit.system != ostree rollback commit`
+1. `microshift run`
+1. System is unhealthy consistently, `boot_counter` falls to `-1`, grub boots 1st commit (sans-MicroShift)
+1. 4th commit (with-MicroShift) is staged
+   - greenboot sets `boot_counter`
+1. 1st commit (sans-MicroShift) shuts down
+1. 4th commit (with-MicroShift) starts
+1. `microshift pre-run`
+   - Delete data and start clean
+     > `prev-boot-commit.system == unhealthy` &&
+     > `current-commit != prev-boot-commit` &&
+     > `current-commit not in health.json` &&
+     > `prev-boot-commit != ostree rollback commit`
+1. `microshift run`
+
+### Test Plan
+
+#### Unit tests
+
+Aiming to write as much as possible in Go, we should strive for maximum testability:
+- Separate code paths for planning (e.g. should it do a backup or restore?)
+  and acting (actually perform backup) - e.g. interface with two methods Plan(), Act()
+  - This will allow testing decisions and actions separately
+  - This will allow easy implementation of --dry-run describing what would happen
+- Due to many interactions with filesystem, filesystem abstraction should be investigated
+  so unit tests can use in-memory filesystem rather than host's to make testing easier
+  and more robust.
+
+#### Functional tests focused on each of the areas (backup, restore, migrate)
+
+Functional tests need to be assessed in terms of effort and impact.
+
+Given that implementation of the enhancement for ostree based systems has significantly
+greater priority than implementation for regular RPM systems,
+there's little incentive to work on exposing functionalities in form of `backup`,
+`restore`, and `migrate` commands.
+By not providing a way to manually trigger these processes, ability to perform functional
+tests in isolation might be severely limited or even impossible without entering territory
+of end to end tests.
+
+#### End to end tests
+
+Sequences from "Workflows in detail" should be implemented in CI.
+
+### Graduation Criteria
+
+Functionality will be GA from the beginning.
+
+- All areas of functionality implemented and available for usage
+- Sufficient test coverage - unit tests (where possible, virtualing/mocking filesystem encouraged),
+  integration tests, e2e tests (CI, QE)
+- End user documentation created
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+N/A
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+N/A
+
+### Version Skew Strategy
+
+See section "allowing and blocking upgrades".
```

Comment:
This section doesn't seem to exist?

Reply:
I think those rules were merged into the "Data migration" section. It is worth calling them out specifically here, though, since we have this section of the template.

Reply:
Fixed it (added a link)